### PR TITLE
feat(AccordionTitle): expose contentWrapper shorthand

### PR DESCRIPTION
**Pull request by silviuavram**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Fixes https://github.com/microsoft/fluent-ui-react/issues/2368.

To change the wrapper slot now it's possible to do:
```jsx
  {
    key: 'p',
    title: {content: 'What is a point?', wrapper: {'data-testid': 'silviu'}},
    content: 'Use the haptic SDD circuit, then you can index the redundant pixel!',
  },
```
And the result will end on the wrapper for the content:
image

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12265)

### CODE REVIEW

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 1]
```diff
@@ -109,31 +114,46 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
...
-          {Box.create(content, {
-            defaultProps: () => ({
-              as: 'span',
-              styles: styles.content
-            })
-          })}
-        </div>
+        {Box.create(wrapper, {
+          defaultProps: () => ({
+            className: cx(AccordionTitle.slotClassNames.contentWrapper, classes.contentWrapper),
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
            className: AccordionTitle.slotClassNames.contentWrapper,
            styles: styles.contentWrapper,
```

As it's a `Box` we should use `styles`

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 2]
```diff
@@ -109,31 +114,46 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
     _.invoke(this.props, 'onFocus', e, this.props);
   };
 
+  handleWrapperOverrides = () => ({
+    onFocus: (e: React.SyntheticEvent) => {
+      this.handleFocus(e);
+    },
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]: There should be a call for `predefinedProps`

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 3]
```diff
@@ -109,31 +114,46 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
     _.invoke(this.props, 'onFocus', e, this.props);
   };
 
+  handleWrapperOverrides = () => ({
+    onFocus: (e: React.SyntheticEvent) => {
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
    onFocus: (e: React.FocusEvent) => {
```

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 4]
```diff
@@ -109,31 +114,46 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
...
+      this.handleFocus(e);
+    },
+    onClick: (e: React.SyntheticEvent) => {
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
    onClick: (e: React.MouseEvent) => {
```

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 5]
```diff
@@ -109,31 +114,46 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
...
+      this.handleClick(e);
+    }
+  });
+
   renderComponent({ ElementType, classes, unhandledProps, styles, accessibility }) {
-    const { contentRef, children, content, indicator } = this.props;
+    const { contentRef, children, content, indicator, wrapper } = this.props;
 
     const contentElement = (
```
Comment (jurokapsiar) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]: nit: let's rename this to contentWrapperElement as content is inside of it

**Comment by silviuavram** [comment]
Thanks for reviewing this, I will begin integrating it!

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 6]
```diff
@@ -35,6 +34,9 @@ export interface AccordionTitleProps extends UIComponentProps, ContentComponentP
   /** If at least one panel needs to stay active and this title does not correspond to the last active one. */
   canBeCollapsed?: boolean;
 
+  /** Shorthand for the content wrapper element. */
+  contentWrapperElement?: ShorthandValue<BoxProps>;
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
  contentWrapper?: ShorthandValue<BoxProps>;
```

I think

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 7]
```diff
@@ -88,7 +91,8 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
     accessibility: accordionTitleBehavior,
     as: 'dt',
     contentRef: _.noop,
-    indicator: {}
+    indicator: {},
+    wrapper: {}
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
    contentWrapper: {}
```

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 8]
```diff
@@ -77,6 +79,7 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
     accordionContentId: PropTypes.string,
     active: PropTypes.bool,
     contentRef: customPropTypes.ref,
+    contentWrapperElement: customPropTypes.wrapperShorthand,
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
    contentWrapper: customPropTypes.wrapperShorthand,
```

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 9]
```diff
@@ -109,31 +113,49 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
...
+    },
+    onClick: (e: React.MouseEvent) => {
+      this.handleClick(e);
+      _.invoke(predefinedProps, 'onClick', e, this.props);
+    }
+  });
+
   renderComponent({ ElementType, classes, unhandledProps, styles, accessibility }) {
-    const { contentRef, children, content, indicator } = this.props;
+    const { contentRef, children, content, indicator, contentWrapperElement } = this.props;
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
    const { contentRef, children, content, indicator, contentWrapper } = this.props;
```

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 10]
```diff
@@ -109,31 +113,49 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
...
-            })
-          })}
-          {Box.create(content, {
-            defaultProps: () => ({
-              as: 'span',
-              styles: styles.content
-            })
-          })}
-        </div>
+        {Box.create(contentWrapperElement, {
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
        {Box.create(contentWrapper, {
```

File: packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx [hunk 11]
```diff
@@ -109,31 +113,49 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
...
+      this.handleClick(e);
+      _.invoke(predefinedProps, 'onClick', e, this.props);
+    }
+  });
+
   renderComponent({ ElementType, classes, unhandledProps, styles, accessibility }) {
-    const { contentRef, children, content, indicator } = this.props;
+    const { contentRef, children, content, indicator, contentWrapperElement } = this.props;
 
     const contentElement = (
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Accordion/AccordionTitle.tsx]:
```suggestion
    const contentWrapperElement = (
```

**Pull request merged**